utils/contracts.ts

```diff
@@ -1,23 +1,35 @@
+import { network, ethers } from 'hardhat'
+
 export const mainnet = {
-  CHAINLINK_PRICE_FEED_REGISTRY: '0x47Fb2585D2C56Fe188D0E6ec628a38b74fCeeeDf' as const,
-  CHAINLINK_USD_QUOTE: '0x0000000000000000000000000000000000000348' as const,
-  WSTETH: '0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0' as const,
-  STETH: '0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84' as const,
-  WETH: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2' as const,
-  DAI: '0x6B175474E89094C44Da98b954EedeAC495271d0F' as const,
-  USDT: '0xdAC17F958D2ee523a2206206994597C13D831ec7' as const,
-  USDC: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48' as const,
-  LDO: '0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32' as const,
-  AGENT: '0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c' as const,
-  SETTLEMENT: '0x9008D19f58AAbD9eD0D60971565AA8510560ab41' as const,
-  VAULT_RELAYER: '0xC92E8bdf79f0507f65a392b0ab4667716BFE0110' as const,
-  DOMAIN_SEPARATOR: '0xc078f884a2676e1345748b1feace7b0abee5d00ecadb6e574dcdd109a63e8943' as const,
+  CHAINLINK_PRICE_FEED_REGISTRY: '0x47Fb2585D2C56Fe188D0E6ec628a38b74fCeeeDf',
+  // Conversion targets: https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/Denominations.sol
+  CHAINLINK_USD_QUOTE: '0x0000000000000000000000000000000000000348',
+  STETH: '0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84',
+  DAI: '0x6B175474E89094C44Da98b954EedeAC495271d0F',
+  USDT: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
+  USDC: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
+  LDO: '0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32',
+  AGENT: '0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c',
+  SETTLEMENT: '0x9008D19f58AAbD9eD0D60971565AA8510560ab41',
+  VAULT_RELAYER: '0xC92E8bdf79f0507f65a392b0ab4667716BFE0110',
+  DOMAIN_SEPARATOR: '0xc078f884a2676e1345748b1feace7b0abee5d00ecadb6e574dcdd109a63e8943',
+  MANAGER: '0xa02FC823cCE0D016bD7e17ac684c9abAb2d6D647'
+}
+
+export const holesky = {
+  AGENT: '0xE92329EC7ddB11D25e25b3c21eeBf11f15eB325d',
+  CHAINLINK_USD_QUOTE: '0x0000000000000000000000000000000000000348',
+  LDO: '0x14ae7daeecdf57034f3E9db8564e46Dba8D97344',
+  STETH: '0x3F1c547b21f65e10480dE3ad8E19fAAC46C95034',
+  CHAINLINK_PRICE_FEED_REGISTRY: '0x9dD5144FFB63a423a18f00a6F6fE53d38058DbFe', // Stub
+  SETTLEMENT: '0xcEE18ed7d5561dC32b8dA5724ffA4d565d2f53ea', // Stub
+  VAULT_RELAYER: '0x36b43D95A6f3C9b6708b57DD5e7C556Dad3C43A0', // Stub
+  DOMAIN_SEPARATOR: '0x3d02cb7d79e60bdbbe81e899cc404908239c89a738b5384b681319579921e414',
+  MANAGER: '0x4BfD997CD068E95577A2917CBCEDD0E99f60f271'
 }
 
-export const goerli = {
-  WETH: '0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6' as const,
-  UNI: '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984' as const,
-  SETTLEMENT: '0x9008D19f58AAbD9eD0D60971565AA8510560ab41' as const,
-  VAULT_RELAYER: '0xC92E8bdf79f0507f65a392b0ab4667716BFE0110' as const,
-  AGENT: '0x4333218072D5d7008546737786663c38B4D561A4' as const,
+export const getContracts = async () => {
+  if (network.name === 'holesky') return holesky
```

Comment:
```javascript
else if (network.name == 'mainnet' || network.name == 'hardhat') return mainnet
throw new Error('Unknown Network')
```
When value not found, do not return default value silently; fail loudly instead.